### ISSUE TITLE
Not possible to configure Neural Computer Stick 2 with OpenVINO + onnxruntime

### ISSUE THREAD

**Issue by thestarivore** [openvinotoolkit/openvino#11725]
##### System information (version)
-  Raspberry PI 4B - Raspbian Bullseye - ARM64
- Intel® Neural Compute Stick 2
- ONNX Runtime version v1.11.0
- OpenVINO 2022.1

##### Detailed description
Hi, I'm trying to get OpenVINO + ONNXRUNTIME  to work on a Raspberry PI 4B with a ARM64 OS (Raspbian Bullseye). But I can't link it to onnxruntime. I'm building onnxruntime and openvino from scratch and they work individually, but I can't get to make onnxruntime recognize the OpenVino EP. Every time I try to follow the official instructions, everything on my system is different than it's described in the instructions and the OpenVINO's setupvars.sh script never works (never finds python). Also if build openvino from scratch i never get the inference_engine and the python folders.

##### Steps to reproduce
Even though I tried many different cmake configurations, the steps I use are:

    ```
        $ git clone https://github.com/openvinotoolkit/openvino.git
        $ cd openvino
        $ git submodule update --init --recursive
        $ chmod +x install_build_dependencies.sh
        $ ./install_build_dependencies.sh
        $ mkdir build && cd build
        $ cmake -DCMAKE_BUILD_TYPE=Release \
        -DENABLE_PYTHON=ON \
        -DPYTHON_EXECUTABLE=`which python3.6` \
        -DPYTHON_LIBRARY=/usr/lib/x86_64-linux-gnu/libpython3.6m.so \
        -DPYTHON_INCLUDE_DIR=/usr/include/python3.6 ..
        $ make --jobs=$(nproc --all)
        $ export PYTHONPATH=$PYTHONPATH:/~/openvino/bin/intel64/Release/lib/python_api/python3.6/
        $ export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:~/openvino/bin/intel64/Release/lib/
    ```

##### Issue submission checklist
Afterwards I execute source setupvars.sh, but it doesn't find python.
I also tried to use cmake 3.14.0 by building it from scratch.

It would be nice to have an already compiled packet like for instance: https://storage.openvinotoolkit.org/repositories/openvino/packages/2021.4.2/l_openvino_toolkit_runtime_raspbian_p_2021.4.752.tgz
The latter is for Rasbian Buster 32bit and it works. I however I'm forced to use Raspbian Bullseye - ARM64.

Any help would be much appreciated.
Thank you in advance.

**Comment by jgespino** [comment]
@thestarivore I am in the process of building OpenVINO 2022.1 from source on Bullseye (64-bit) on my raspberry pi 4. Will you just be using Python?

In the meantime, could you try adding `-DCMAKE_INSTALL_PREFIX=/opt/intel/openvino` to your cmake command and rebuilding. It should not take very long. Then run `sudo make install` and source the environment `source /opt/intel/openvino/setupvars.sh`.

This is mentioned in this [guide](https://www.intel.com/content/www/us/en/support/articles/000057005/boards-and-kits.html) for Raspbian OS 32-bit.

**Comment by thestarivore** [comment]
@jgespino thanks a lot, with your instructions it worked (I had to also recompile onnxruntime). Yeah I'll be using python and I need to use onnxruntime+openvino and onnx.

The only thing missing now is the model optimizer. If I understood correctly it should be in:
   `<INSTALL_DIR>/deployment_tools/model_optimizer directory.
`
But all I can see is:

> starivore@raspberrypi:~ $ ls /opt/intel/openvino
   docs  install_dependencies  python  runtime  samples  setupvars.sh  tools

Do I need to install it separately?

**Comment by jgespino** [comment]
@thestarivore Correct, the error indicated that the NCS 2 device has ran out of memory. Would you mind sharing your code and model for me to reproduce from my side?

**Comment by thestarivore** [comment]
@jgespino Sure, here is the model:
[model.zip](https://github.com/openvinotoolkit/openvino/files/8791379/model.zip)
The input an be a random tensor [32, 96, 80, 80]

I would like to fall back to CPU when the inference on NCS2 with openvino fails. But when I get this error (X_LINK_OUT_OF_MEMORY), the whole script crashes. 
Is there a way to catch this error and start another inference using the CPU? Because try except doesn't work. Thanks in advance!

**Comment by brmarkus** [comment]
What do you mean with "having a problem that happens after ~50 executions"? Do you try to instantiate the exec-net 50 times, i.e. you want to run that many number of independent instances, have many models loaded in parallel?

**Comment by jgespino** [comment]
@thestarivore Have you tried decreasing the input to something smaller than batch 32? I was able to run benchmark_app with your model using input shape up to [22,96,80,80].

> Is there a way to catch this error and start another inference using the CPU? Because try except doesn't work. Thanks in advance!

I'm not sure if catching the error is possible as the error/messages are coming from the Myriad Plugin driver. I recommend taking a look at the Myriad Plugin code:
https://github.com/openvinotoolkit/openvino/tree/master/src/plugins/intel_myriad/myriad_plugin

Regards,
Jesus

**Comment by jgespino** [comment]
Closing, feel free to start a new issue if additional assistance is needed.

**Comment by LL223851** [comment]
> ##### System information (version)
> * Raspberry PI 4B - Raspbian Bullseye - ARM64
> * Intel® Neural Compute Stick 2
> * ONNX Runtime version v1.11.0
> * OpenVINO 2022.1
> 
> ##### Detailed description
> Hi, I'm trying to get OpenVINO + ONNXRUNTIME to work on a Raspberry PI 4B with a ARM64 OS (Raspbian Bullseye). But I can't link it to onnxruntime. I'm building onnxruntime and openvino from scratch and they work individually, but I can't get to make onnxruntime recognize the OpenVino EP. Every time I try to follow the official instructions, everything on my system is different than it's described in the instructions and the OpenVINO's setupvars.sh script never works (never finds python). Also if build openvino from scratch i never get the inference_engine and the python folders.
> 
> ##### Steps to reproduce
> Even though I tried many different cmake configurations, the steps I use are:
> 
> ```
> ```
>     $ git clone https://github.com/openvinotoolkit/openvino.git
>     $ cd openvino
>     $ git submodule update --init --recursive
>     $ chmod +x install_build_dependencies.sh
>     $ ./install_build_dependencies.sh
>     $ mkdir build && cd build
>     $ cmake -DCMAKE_BUILD_TYPE=Release \
>     -DENABLE_PYTHON=ON \
>     -DPYTHON_EXECUTABLE=`which python3.6` \
>     -DPYTHON_LIBRARY=/usr/lib/x86_64-linux-gnu/libpython3.6m.so \
>     -DPYTHON_INCLUDE_DIR=/usr/include/python3.6 ..
>     $ make --jobs=$(nproc --all)
>     $ export PYTHONPATH=$PYTHONPATH:/~/openvino/bin/intel64/Release/lib/python_api/python3.6/
>     $ export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:~/openvino/bin/intel64/Release/lib/
> ```
> ```
> 
> ##### Issue submission checklist
> Afterwards I execute source setupvars.sh, but it doesn't find python. I also tried to use cmake 3.14.0 by building it from scratch.
> 
> It would be nice to have an already compiled packet like for instance: https://storage.openvinotoolkit.org/repositories/openvino/packages/2021.4.2/l_openvino_toolkit_runtime_raspbian_p_2021.4.752.tgz The latter is for Rasbian Buster 32bit and it works. I however I'm forced to use Raspbian Bullseye - ARM64.
> 
> Any help would be much appreciated. Thank you in advance.

-DPYTHON_EXECUTABLE=`which python3.6` \
to this
-DCYTHON_EXECUTABLE=$(which cython) \
